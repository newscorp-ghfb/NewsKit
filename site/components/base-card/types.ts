import {CardProps} from 'newskit';

export interface BaseCardProps extends Omit<CardProps, 'children'> {
  children?: React.ReactNode;
  title?: string;
  description?: string | React.ReactNode;
}
