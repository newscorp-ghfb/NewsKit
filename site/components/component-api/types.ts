import {TableRow} from '../table';

export interface ComponentAPIProps {
  components: SingleComponentAPIProps[];
}

export interface SingleComponentAPIProps {
  title?: string;
  summary?: string | JSX.Element;
  propsSummary?: string;
  argsSummary?: string;
  overridesSummary?: string;
  propsRows?: PropsRowsProps[];
  propsFooter?: React.ReactNode;
  argsRows?: ArgumentsRowsProps[];
  argsFooter?: React.ReactNode;
  overridesRows?: OverridesRowsProps[];
  overridesFooter?: React.ReactNode;
}

interface CommonProps extends TableRow {
  type: string | string[];
  description: string | JSX.Element;
  default?: string | string[];
  required?: boolean;
}

export interface PropsRowsProps extends CommonProps {
  name: string;
}

export interface OverridesRowsProps extends CommonProps {
  attribute: string;
}
export interface ArgumentsRowsProps extends CommonProps {
  order: number;
}
