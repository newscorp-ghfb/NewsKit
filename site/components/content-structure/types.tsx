import {MQ} from 'newskit';

export interface ContentBaseProps {
  headline?: string | React.ReactElement;
  description?: string | React.ReactElement;
  toc?: string;
  id?: string;
  children?: React.ReactNode;
  childrenColSpan?: number;
  hideBottomSpacing?: boolean;
  overrides: {
    headline?: {
      as: 'h2' | 'h3' | 'h4';
      typographyPreset: MQ<string>;
      stylePreset: MQ<string>;
      spaceStack?: MQ<string>;
    };
    description?: {
      typographyPreset: MQ<string>;
      stylePreset: MQ<string>;
      spaceStack?: MQ<string>;
    };
    separator?: {
      spaceStack: MQ<string>;
    };
  };
  showSeparator?: boolean;
}

export type ContentPrimaryProps = Omit<ContentBaseProps, 'overrides'>;

export type ContentSecondaryProps = Omit<
  ContentBaseProps,
  'overrides' | 'toc' | 'id'
>;

export type ContentTertiaryProps = Omit<
  ContentBaseProps,
  'overrides' | 'toc' | 'id'
>;

export type ContentSectionProps = {
  children: React.ReactNode;
  sectionName?: string;
  toc?: string;
  id?: string;
};
