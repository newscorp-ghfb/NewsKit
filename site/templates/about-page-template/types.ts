import {TemplateProps} from '../types';
import {LayoutProps} from '../../components/layout';
import {OnwardJourneySectionProps} from '../template-sections';

export interface AboutPageTemplateProps
  extends Omit<TemplateProps, 'featureCard'> {
  layoutProps: LayoutProps;
  featureCard?: Partial<OnwardJourneySectionProps>;
}
