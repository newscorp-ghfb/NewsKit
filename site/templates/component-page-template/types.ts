import {LayoutProps} from '../../components/layout';
import {TemplateProps} from '../types';
import {
  InteractiveDemoSectionProps,
  AnatomySectionProps,
  OptionsSectionProps,
  StatesSectionProps,
  BehaviorsSectionProps,
  CodeExamplesSectionProps,
  UsageSectionProps,
  AccessibilitySectionProps,
  SEOSectionProps,
  ComponentAPISectionProps,
  ComplianceSectionProps,
  RelatedComponentsSectionProps,
  OnwardJourneySectionProps,
  CommonSectionProps,
} from '../template-sections';

export interface ComponentPageTemplateProps
  extends Omit<TemplateProps, 'featureCard' | 'children'> {
  layoutProps: LayoutProps;
  componentDefaultsKey: string;
  interactiveDemo?: InteractiveDemoSectionProps;
  anatomy?: AnatomySectionProps;
  options?: OptionsSectionProps;
  states?: StatesSectionProps;
  behaviors?: BehaviorsSectionProps;
  codeExamples?: CodeExamplesSectionProps;
  usage?: UsageSectionProps;
  accessibility?: AccessibilitySectionProps;
  seo?: SEOSectionProps;
  componentAPI?: ComponentAPISectionProps;
  compliance?: ComplianceSectionProps;
  related?: RelatedComponentsSectionProps;
  featureCard?: Partial<OnwardJourneySectionProps>;
  children?: React.ReactNode;
  commonSection?: CommonSectionProps;
}
