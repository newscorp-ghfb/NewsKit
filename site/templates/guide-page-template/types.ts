import {TemplateProps} from '../types';
import {LayoutProps} from '../../components/layout';
import {OnwardJourneySectionProps} from '../template-sections';

export interface GuidePageTemplateProps
  extends Omit<TemplateProps, 'featureCard'> {
  layoutProps: LayoutProps;
  featureCard?: Partial<OnwardJourneySectionProps>;
}
