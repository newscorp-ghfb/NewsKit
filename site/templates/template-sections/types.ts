export interface IntroductionText {
  title?: string;
  introduction?: string | React.ReactElement;
}
