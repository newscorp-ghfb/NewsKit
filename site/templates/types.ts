import {
  IntroductionSectionProps,
  OnwardJourneySectionProps,
} from './template-sections';

export interface TemplateProps extends IntroductionSectionProps {
  children: React.ReactNode;
  headTags: {
    title: string;
    description: string;
    imageUrl?: string;
    alt?: string;
    width?: string;
    height?: string;
  };
  featureCard?: OnwardJourneySectionProps;
}
