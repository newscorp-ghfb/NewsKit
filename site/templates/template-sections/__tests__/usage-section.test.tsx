import {UsageKind} from '../../../components/usage-card/types';
import {UsageSection, UsageSectionProps} from '..';
import {renderToFragmentWithTheme} from '../../../utils/test-utils';

jest.mock('../../../components/media-list/media-list.tsx');
jest.mock(
  'newskit',
  require('../../../utils/test-utils').mockNewsKitComponents(
    'Grid',
    'Cell',
    'InlineMessage',
  ),
);

describe('UsageSection', () => {
  test('renders section as expected', () => {
    const props: UsageSectionProps = {
      introduction: 'states introduction text goes here',
      cards: [
        {
          description: 'description text 1',
          media: {
            src: 'media 1 src',
            alt: 'media 1',
          },
          kind: UsageKind.DO,
        },
        {
          description: 'description text 2',
          media: {
            src: 'media 2 src',
            alt: 'media 2',
          },
          kind: UsageKind.DONT,
        },
      ],
      notice: 'Inline Message',
    };

    const fragment = renderToFragmentWithTheme(UsageSection, props);
    expect(fragment).toMatchSnapshot();
  });
});
