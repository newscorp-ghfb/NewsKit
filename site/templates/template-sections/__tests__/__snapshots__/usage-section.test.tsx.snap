// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsageSection renders section as expected 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-top: -24px;
  padding-top: 24px;
}

@media screen and (max-width: 767px) {
  .emotion-1 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-1 {
    margin-bottom: 32px;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 0;
  background-color: transparent;
}

.emotion-3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

@media screen and (max-width: 767px) {
  .emotion-4 {
    font-family: "Poppins",sans-serif;
    font-size: 28px;
    line-height: 31.5px;
    font-weight: 700;
    letter-spacing: 0;
    padding: 0.5px 0px;
  }

  .emotion-4::before {
    content: '';
    margin-bottom: -0.2155em;
    display: block;
  }

  .emotion-4::after {
    content: '';
    margin-top: -0.2125em;
    display: block;
  }
}

@media screen and (min-width: 768px) {
  .emotion-4 {
    font-family: "Poppins",sans-serif;
    font-size: 32px;
    line-height: 36px;
    font-weight: 700;
    letter-spacing: 0;
    padding: 0.5px 0px;
  }

  .emotion-4::before {
    content: '';
    margin-bottom: -0.2155em;
    display: block;
  }

  .emotion-4::after {
    content: '';
    margin-top: -0.2125em;
    display: block;
  }
}

.emotion-5 {
  display: inline;
  font: inherit;
  margin: 0;
  color: #09111C;
}

.emotion-5 svg {
  fill: #09111C;
}

@media screen and (max-width: 1023px) {
  .emotion-6 {
    margin-bottom: 48px;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-6 {
    margin-bottom: 80px;
  }
}

.emotion-7 {
  margin: 0;
  color: #2E3F54;
  font-family: "Poppins",sans-serif;
  font-size: 20px;
  line-height: 30px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
}

.emotion-7 svg {
  fill: #2E3F54;
}

.emotion-7::before {
  content: '';
  margin-bottom: -0.403em;
  display: block;
}

.emotion-7::after {
  content: '';
  margin-top: -0.4em;
  display: block;
}

<div
    data-comp="NewsKit Cell"
    data-props="{
  \\"xs\\": 12
}"
    data-testid="Cell"
  >
    <section
      class="emotion-0"
      data-toc-indexed="Usage"
      id="usage"
    >
      <div
        data-comp="NewsKit Grid"
        data-props="{
  \\"xsMargin\\": \\"space000\\",
  \\"xsRowGutter\\": \\"space000\\"
}"
        data-testid="Grid"
      >
        <div
          data-comp="NewsKit Cell"
          data-props="{
  \\"xs\\": 12,
  \\"md\\": 8,
  \\"lg\\": 10,
  \\"xl\\": 8,
  \\"mdOffset\\": 1
}"
          data-testid="Cell"
        >
          <div
            class="emotion-1"
          >
            <div
              class="emotion-2"
            >
              <div
                class="emotion-3"
              >
                <section
                  class="emotion-4"
                >
                  <h2
                    class="nk-headline-heading emotion-5"
                  >
                    Usage
                  </h2>
                </section>
              </div>
            </div>
          </div>
          <div
            class="emotion-6"
          >
            <p
              class="emotion-7"
            >
              states introduction text goes here
            </p>
          </div>
        </div>
        <div
          data-comp="NewsKit Cell"
          data-props="{
  \\"xs\\": 12,
  \\"mdOffset\\": 1,
  \\"md\\": 10,
  \\"xl\\": 8
}"
          data-testid="Cell"
        />
        <div
          data-comp="NewsKit Cell"
          data-props="{
  \\"xs\\": 12,
  \\"md\\": 10,
  \\"xl\\": 8,
  \\"mdOffset\\": 1
}"
          data-testid="Cell"
        >
          <div>
            MediaList Component with props:{
  "gridProps": {
    "xsRowGutter": "space050"
  },
  "cardType": "usage",
  "layout": "2-span",
  "cards": [
    {
      "description": "description text 1",
      "media": {
        "src": "media 1 src",
        "alt": "media 1"
      },
      "kind": "do"
    },
    {
      "description": "description text 2",
      "media": {
        "src": "media 2 src",
        "alt": "media 2"
      },
      "kind": "dont"
    }
  ]
}
          </div>
          <div
            data-comp="NewsKit InlineMessage"
            data-props="{
  \\"icon\\": {
    \\"type\\": {
      \\"stylePresets\\": {
        \\"iconPositive\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.inkPositive}}\\"
          }
        },
        \\"iconNegative\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.inkNegative}}\\"
          }
        },
        \\"iconSocialTwitter\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.socialTwitter}}\\"
          }
        },
        \\"iconSocialGitHub\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.socialGithub}}\\"
          }
        },
        \\"iconSocialFacebook\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.socialFacebook}}\\"
          }
        },
        \\"iconSocialInstagram\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.socialInstagram}}\\"
          }
        },
        \\"iconSocialYoutube\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.socialYoutube}}\\"
          }
        },
        \\"iconSocialWhatsApp\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.socialWhatsapp}}\\"
          }
        },
        \\"iconSocialReddit\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.socialReddit}}\\"
          }
        },
        \\"iconDefault\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.inkBase}}\\"
          }
        },
        \\"indeterminateProgressIndicatorPrimary\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.inkBrand010}}\\"
          }
        },
        \\"indeterminateProgressIndicatorSecondary\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.inkBase}}\\"
          }
        },
        \\"indeterminateProgressIndicatorNegative\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.inkNegative}}\\"
          }
        },
        \\"indeterminateProgressIndicatorPositive\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.inkPositive}}\\"
          }
        },
        \\"indeterminateProgressIndicatorInverse\\": {
          \\"base\\": {
            \\"iconColor\\": \\"{{colors.inkInverse}}\\"
          }
        }
      }
    },
    \\"key\\": null,
    \\"ref\\": null,
    \\"props\\": {
      \\"overrides\\": {
        \\"size\\": \\"iconSize020\\"
      }
    },
    \\"_owner\\": null,
    \\"_store\\": {}
  },
  \\"role\\": \\"region\\",
  \\"aria-label\\": \\"usage notice\\",
  \\"overrides\\": {
    \\"marginBlockStart\\": \\"space070\\"
  }
}"
            data-testid="InlineMessage"
          >
            Inline Message
          </div>
        </div>
      </div>
    </section>
  </div>
  @media screen and (max-width: 767px) {
  .emotion-0 {
    margin-top: 64px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    margin-top: 80px;
  }
}

@media screen and (max-width: 767px) {
  .emotion-0 {
    margin-bottom: 40px;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    margin-bottom: 80px;
  }
}

.emotion-1 {
  border-style: solid;
  border-color: #DEE2E5;
  border-width: 1px;
  border-width: 0px;
  margin: 0;
  border-top-width: 1px;
  width: 100%;
}

<div
    data-comp="NewsKit Cell"
    data-props="{
  \\"xs\\": 12,
  \\"md\\": 10,
  \\"xl\\": 8,
  \\"mdOffset\\": 1
}"
    data-testid="Cell"
  >
    <div
      class="emotion-0"
    >
      <hr
        aria-hidden="true"
        class="emotion-1"
        data-testid="divider"
      />
    </div>
  </div>
</DocumentFragment>
`;
